### DISCUSSION TITLE
OSCAL 1.0.3 Release

### DISCUSSION THREAD

**Discussion by david-waltermire** [usnistgov/OSCAL#1245] [Releases]
**Note: This release contains defective JSON schemas that are missing required regular expressions. This has been corrected in the [OSCAL 1.0.4 release](https://github.com/usnistgov/OSCAL/releases/tag/v1.0.4). Please use the 1.0.4 release instead of this release.**

## What's Changed

- #737, #1208 Updated link checker for generate website workflow and improve workflows. (PR #1231) @aj-stein-nist
- #1062 Clarifying system security plan cross-references (PR #1167) @Rene2mt, @aj-stein-nist, @david-waltermire-nist
- #1127, #1186 Updated to latest Metaschema toolchain (PR #1161) @wendellpiez, @david-waltermire-nist, aj-stein-nist
  - Properly encode high-order Unicode characters in schemas (PR usnistgov/metaschema#183)
  - usnistgov/metaschema#181 Rolled back token datatype to avoid problematic character references
  - usnistgov/metaschema#182 Fixed double-escaped RegEx patterns in output JSON schema causing issues with some RegEx flavors. @wendellpiez
  - usnistgov/metaschema#188 Added support for durations. @wendellpiez, @david-waltermire-nist
  - Adjusted data types in generated XML and JSON schemas. @wendellpiez, @david-waltermire-nist
  - Corrected pruning of unused definitions during XML and JSON schema generation. (PR usnistgov/metaschema#199) @wendellpiez, @aj-stein-nist, @david-waltermire-nist
- #1193 Grammar fixes to `assessment-common` metaschema only. (PR #1224) @guyzyl
- #1194 Clarified the semantics of implemented-requirement in a component definition as only a suggestion of how to implement. (PR #1232) @david-waltermire-nist
- #1198 Replaced master with main in the git-cheat-sheet.md file' (#1200) @iMichaela
- #1206 Clarified use of set-parameter and how to determine the effective value. (PR #1234) @david-waltermire-nist
- #1209, #1212 Added Issue Triage automation to project. (PR #1210) @aj-stein-nist
- #1211 Clarified use of profile and catalog in import-profile/@href. (PR #1227) @david-waltermire-nist
- #1216 Ensured GitHub Actions Installs Python Dependencies Like Local Docker Environment. (PR usnistgov/oscal-content#103)
- #1220 Corrected metadata/location/address/country regex (PR #1226) @david-waltermire-nist
- Bump [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6 in /build (PR #1195) @dependabot
- Changed GitHub links to point to `main` branch instead of `master` branch. (PR #1225) @guyzyl
- Normalized enumerated namespace values to use http prefix consistently (PR #1233) @david-waltermire-nist
  - Normalized enumerated namespace values to use http prefix consistently; created deprecation entries where needed.
  - Updated examples to be consistent with namespace values.
  - Added deprecation entries for old values.
- Fixed typo in the word "separately" (PR #1235) @galtm
- Boolean handling workaround for workflow_call vs workflow_dispatch bug. (PR #1236) @aj-stein-nist
- Improved actions workflow and steps for package release automation. (PR #1240) @aj-stein-nist, @david-waltermire-nist
- Removed unneeded Ruby dependencies. @david-waltermire-nist
- Reordered tools listing alphabetically. @david-waltermire-nist
- Add Ignyte to tools page. (PR ##1197) @caseykulasa

## Notable Changes

To fix a number of data type related issues, the underlying type system used in the generated OSCAL XML and JSON schemas was replaced. This change resulted in different names for simple and complex types for data types in XML schemas and some adjustments in data type definitions in JSON schemas. This may cause some issues with schema binding approaches that generate code from the XML or JSON schemas. In such instances, you may need to further customize your binding configurations or make some code adjustments resulting from differently generated code.

**Full Changelog**: https://github.com/usnistgov/OSCAL/compare/v1.0.2...v1.0.3

<hr /><em>This discussion was created from the release <a href='https://github.com/usnistgov/OSCAL/releases/tag/v1.0.3'>OSCAL 1.0.3 Release</a>.</em>

**Comment by andrewliddell-cyber** [comment]
Hello, @david-waltermire, is the OSCAL-CLI build compatible with JDK 25.0.2? Or does it specifically require JDK 17? I'm encountering "PKIX path building failed unable to find valid certification path to requested target" when trying to install it using Maven.

**Reply by iMichaela** [reply]
> Hello, @david-waltermire, is the OSCAL-CLI build compatible with JDK 25.0.2? Or does it specifically require JDK 17? I'm encountering "PKIX path building failed unable to find valid certification path to requested target" when trying to install it using Maven.

@andrewliddell-cyber - The latest OSCAL release is 1.2.1 https://github.com/usnistgov/OSCAL/releases/tag/v1.2.1. The oscal-cli repo   is here: https://github.com/usnistgov/oscal-cli. I have not tested the build with any version newer than JDK 21

**Comment by andrewliddell-cyber** [comment]
Thank you @iMichaela, is there somewhere to send "PKIX path building failed unable to find valid certification path to requested target" error message for resolution?

**Reply by iMichaela** [reply]
> Thank you @iMichaela, is there somewhere to send "PKIX path building failed unable to find valid certification path to requested target" error message for resolution?

I must ask /understand why you want to use such an old version of OSCAL? DO you mind connect with us via email : oscal@nist.gov so we can better understand and guide. I am assuming answering my question here might not be something you are comfortable with.

**Reply by andrewliddell-cyber** [reply]
Hi Dr. Iorga, happy to provide context. I'm not intentionally trying to use an old version of OSCAL. The current instructions for installing and getting this running on my local machine are what are causing me to see this error. I've emailed more details to the address provided. Thank you!

**Reply by iMichaela** [reply]
> Hi Dr. Iorga, happy to provide context. I'm not intentionally trying to use an old version of OSCAL. The current instructions for installing and getting this running on my local machine are what are causing me to see this error. I've emailed more details to the address provided. Thank you!

I am assuming you are referring to the oscal-cli 1.0.3 version, and not OSCAL 1.0.3.

For example, oscal-cli  1.0.4 (see below),  was generated using the OSCAL 1.1.3 version. :

**oscal-cli 1.0.4** built at 2025-03-12 00:23 from branch main (5cfe61c) at https://github.com/usnistgov/oscal-cli.git
liboscal-java v3.0.3 built at 2025-03-12 00:10 from branch main (f2ca876) at https://github.com/usnistgov/liboscal-java.git
**oscal v1.1.3** built at 2025-03-12 00:10 from branch b123c11bd12c8b8f1bcc8bf85763e5775c0423e9 (b123c11) at https://github.com/usnistgov/OSCAL.git
metaschema-java 0.12.2 built at 2023-09-19T20:12:43+0000 from branch f50c0658054175fd2d72fb8755404e8a998e0fb3 (f50c065) at https://github.com/usnistgov/metaschema-java
metaschema v0.9.0 built at 2023-09-19T20:12:43+0000 from branch a36f579e1e30abb2263895242cdbd2cf4bd29513 (a36f579) at https://github.com/usnistgov/metaschema

If the guidance is confusing , we need to fix it. I will look for your email with the details of the error.

**Comment by iMichaela** [comment]
The PKIX error is not caused by the pom.xml file. It is a Java SSL/TLS handshake failure that occurs when Maven tries to download dependencies over HTTPS and Java cannot verify the server's certificate chain. Your network proxy (e.g. Zscaler, BlueCoat, Palo Alto) intercepts HTTPS traffic and re-signs it with its own certificate. Java's bundled truststore (cacerts) doesn't know that certificate, so it rejects the connection entirely. 

I suggest you should import your network proxy certificate - this would be a permanent fix for you.

```bash
# Step 1: Get your proxy certificate — ask your IT/network team for the .crt file OR export it from your browser:
# Chrome/Edge → padlock icon → Certificate → Details → Export as .crt

# Step 2: Import it into Java's truststore
keytool -import \
  -alias corporate-proxy \
  -keystore "$JAVA_HOME/lib/security/cacerts" \
  -file your-proxy-certificate.crt \
  -storepass changeit
  ```
